### DISCUSSION TITLE
modbus tcp

### DISCUSSION THREAD

**Discussion by basking-in-the-sun2000** [basking-in-the-sun2000/solar-logger#49] [General]
This is why I like that the logger is using Modbus TCP (referred to operational technology in the video below) all the time. It locks a second connection from occurring. Also, probably the reason Huawei started disabling TCP by default.

It is highly inconvenient since you must stop the logger whenever you want to use the app, but I've thought of that as a security feature. The external logger works around this since the inverter pushes the data to them. 

[![SN about modbus](http://img.youtube.com/vi/fLvLhlA32CI/0.jpg)](http://www.youtube.com/watch?v=fLvLhlA32CI&t=2840s "Modbus security")

**Comment by andysu** [comment]
Unfortunately, the inverter uses undocumented authentication mechanisms when connecting via TСP modbus. Therefore, I do not know how to use the functions 0x41 in this case. While everything works when connected via RTU modbus. (I was able to download the log files and optimizer work files.)
